### ISSUE TITLE
SSO broken after upgrade from 26.0.3.2 to 26.0.4.2. Invalid metadata

### ISSUE THREAD

**Issue by gonzalo** [nextcloud/user_saml#755]
### Steps to reproduce
1. Upgrade to latest minor nextcloud version 26.0.4.2
2. Use our sso to login
3. Get a "User not provisioned" message
4. Check nextclloud logs, get an error ·Invalid SAML Response. Not match the saml-schema-protocol-2.0.xsd· 
5. Go to Administraton settings -> SSO 
6. Click on download metadata XML https://myinstalnce/apps/user_saml/saml/metadata?idp=1 -> Nextcloud page "Internal server error. 
7. Same error with curl

### Expected behaviour
Normal login 

### Actual behaviour
Tested on two different test nextcloud instances. Same behaviour.
- First one upgraded from 25.0.4 to 25.0.9 and then to 26.0.4, now fails
- Second one upgraded from 26.0.3. to 26.0.4. It breaks. Reverted to 26.0.3 (without touching DB) and now is working

- Plugin was on 5.2.1 since it was released

It seems for me that validator is broken/incompatible with latest nextcloud version? It doesn't validate the answer from our CAS but I'm also unable to generate the Metadata file to configure it again.

### Server configuration
**Operating system**: Debian 11

**Web server:**Apache 2

**Database:**Maria DB

**PHP version:**php 8.1

**Nextcloud version:** 26.0.4

**Where did you install Nextcloud from:** Manual install + upgrade from updater

**List of activated apps:**

```
  - activity: 2.18.0
  - admin_audit: 1.16.0
  - announcementcenter: 6.6.1
  - circles: 26.0.0
  - cloud_federation_api: 1.9.0
  - comments: 1.16.0
  - contacts: 5.3.2
  - contactsinteraction: 1.7.0
  - dashboard: 7.6.0
  - dav: 1.25.0
  - federatedfilesharing: 1.16.0
  - federation: 1.16.0
  - files: 1.21.1
  - files_accesscontrol: 1.16.0
  - files_pdfviewer: 2.7.0
  - files_rightclick: 1.5.0
  - files_sharing: 1.18.0
  - files_trashbin: 1.16.0
  - files_versions: 1.19.1
  - logreader: 2.11.0
  - lookup_server_connector: 1.14.0
  - nextcloud_announcements: 1.15.0
  - notifications: 2.14.0
  - oauth2: 1.14.1
  - password_policy: 1.16.0
  - photos: 2.2.0
  - privacy: 1.10.0
  - provisioning_api: 1.16.0
  - recommendations: 1.5.0
  - related_resources: 1.1.0
  - richdocuments: 8.0.2
  - serverinfo: 1.16.0
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - support: 1.9.0
  - survey_client: 1.14.0
  - systemtags: 1.16.0
  - text: 3.7.2
  - theming: 2.1.1
  - twofactor_backupcodes: 1.15.0
  - updatenotification: 1.16.0
  - user_saml: 5.2.1
  - user_status: 1.6.0
  - viewer: 1.10.0
  - workflowengine: 2.8.0
```

**Nextcloud configuration:**

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            ""
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "tempdirectory": "xxxxxx\/temp",
        "version": "26.0.4.2",
        "overwrite.cli.url": "https:\/\/myinstance.com",
        "htaccess.RewriteBase": "\/",
        "dbtype": "mysql",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "filelocking.enabled": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "dbindex": 0
        },
        "maintenance": false,
        "theme": "",
        "loglevel": 3,
        "installed": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "updater.release.channel": "stable",
        "lost_password_link": "disabled",
        "defaultapp": "files",
        "allow_user_to_change_display_name": false,
        "hide_login_form": false,
        "skeletondirectory": "",
        "templatedirectory": "",
        "session_lifetime": 28800,
        "profile.enabled": false,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap"
        ],
        "config_is_read_only": true
    }
}
```

### Client configuration
**Browser:**Chrome / firefox

**Operating system:** Mac / Linux

#### Nextcloud log (data/owncloud.log)
```
{"reqId":"rXtdaf7jpfFDs5nKQ8m9","level":3,"time":"2023-07-21T11:23:51+00:00","remoteAddr":"xxxxxxxxxxx","user":"--","app":"PHP","method":"POST","url":"/apps/user_saml/saml/acs","message":"DOMDocument::schemaValidate(): Invalid Schema at /aplica/nextcloud/www/nextcloud/apps/user_saml/3rdparty/vendor/onelogin/php-saml/src/Saml2/Utils.php#153","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36","version":"26.0.4.2","data":{"app":"PHP"}}
{"reqId":"rXtdaf7jpfFDs5nKQ8m9","level":4,"time":"2023-07-21T11:23:51+00:00","remoteAddr":"xxxxxxxxxxx","user":"--","app":"user_saml","method":"POST","url":"/apps/user_saml/saml/acs","message":"invalid_response","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36","version":"26.0.4.2","data":{"app":"user_saml"}}
{"reqId":"rXtdaf7jpfFDs5nKQ8m9","level":4,"time":"2023-07-21T11:23:51+00:00","remoteAddr":"xxxxxxxxxxx","user":"--","app":"user_saml","method":"POST","url":"/apps/user_saml/saml/acs","message":"Invalid SAML Response. Not match the saml-schema-protocol-2.0.xsd","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36","version":"26.0.4.2","data":{"app":"user_saml"}}```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```

**Comment by dav2000** [comment]
+1 

I am facing exactly the same issue with Nextcloud 27.0.1 and plugin version 5.2.1.

Postgresql 14.8 and PHP 8.2.8

`{"reqId":"lKGqVUilyZiN1hhj0vDR","level":3,"time":"2023-07-21T14:12:17+00:00","remoteAddr":"xxxxxxxx","user":"nc-admin","app":"PHP","method":"GET","url":"/apps/user_saml/saml/metadata?idp=1","message":"DOMDocument::schemaValidate(): Invalid Schema at /config/www/nextcloud/apps/user_saml/3rdparty/vendor/onelogin/php-saml/src/Saml2/Utils.php#153","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/115.0","version":"27.0.1.2","data":{"app":"PHP"}}

{"reqId":"lKGqVUilyZiN1hhj0vDR","level":3,"time":"2023-07-21T14:12:17+00:00","remoteAddr":"xxxxxxxx","user":"nc-admin","app":"index","method":"GET","url":"/apps/user_saml/saml/metadata?idp=1","message":"Invalid SP metadata: invalid_xml","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/115.0","version":"27.0.1.2","exception":{"Exception":"OneLogin\\Saml2\\Error","Message":"Invalid SP metadata: invalid_xml","Code":3,"Trace":[{"file":"/app/www/public/lib/private/AppFramework/Http/Dispatcher.php","line":230,"function":"getMetadata","class":"OCA\\User_SAML\\Controller\\SAMLController","type":"->"},{"file":"/app/www/public/lib/private/AppFramework/Http/Dispatcher.php","line":137,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/app/www/public/lib/private/AppFramework/App.php","line":183,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/app/www/public/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/app/www/public/lib/base.php","line":1071,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/app/www/public/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/config/www/nextcloud/apps/user_saml/lib/Controller/SAMLController.php","Line":297,"CustomMessage":"--"}}
`

**Comment by blizzz** [comment]
> So I suppose is some kind of incompatibility related to the latest minor versions of nextcloud 25.0.9 26.0.4 and probably 27.0.1

That's accurate. Workaround: disable XML validation for now. Fix candidate in #754
